packages/bot/src/scratch/hooks/toolbox.js

```diff
@@ -80,122 +80,135 @@ Blockly.Toolbox.prototype.populate_ = function (newTree) {
  * @private
  */
 Blockly.Toolbox.prototype.showCategory_ = function (category_id) {
-    let flyout_content;
+    let flyout_content = [];
+    let is_search = false;
 
     if (category_id === 'search') {
-        let search_term = document.getElementById('search_input').value;
-        const all_variables = this.flyout_.workspace_.getVariablesOfType('');
+        const search_term = document.getElementById('search_input').value.toUpperCase();
+        const all_variables = Blockly.derivWorkspace.getVariablesOfType('');
         const all_procedures = Blockly.Procedures.allProcedures(Blockly.derivWorkspace);
 
+        is_search = true;
+
         if (search_term.length <= 1) {
             flyout.setVisibility(false);
             return;
         }
 
-        flyout_content = {
-            type  : 'search',
-            blocks: [],
-        };
+        const { derivWorkspace: { toolbox_: { categoryMenu_: { categories_ } } } } = Blockly;
+        // eslint-disable-next-line consistent-return
+        const block_contents = categories_
+            .filter(category => !category.has_child_category_)
+            .map(category => {
+                let contents = category.getContents();
 
-        if (typeof search_term === 'string') {
-            search_term = search_term.trim().toLowerCase();
-            search_term = search_term.split(' ');
-        }
+                if (typeof contents === 'string') {
+                    const fnToApply = Blockly.derivWorkspace.getToolboxCategoryCallback(contents);
+                    contents = fnToApply(Blockly.derivWorkspace);
+                }
 
-        const blocks = Blockly.Blocks;
-        Object.keys(blocks).forEach(blockKey => {
-            let keywords = ` ${blockKey}`;
-            const block = blocks[blockKey];
+                const only_block_contents = contents.filter(content => content.tagName.toUpperCase() === 'BLOCK');
+                return only_block_contents;
+            })
+            .flat();
+
+        block_contents.forEach(block_content => {
+            const block_type = block_content.getAttribute('type');
+            const block = Blockly.Blocks[block_type];
             const block_meta = block.meta instanceof Function && block.meta();
-            const block_definition = block.definition instanceof Function && block.definition();
+            const block_definitions = block.definition instanceof Function && block.definition();
 
             if (!block_meta) {
                 return;
             }
+            
+            // block_name matched
```

Comment:
When commenting, please clarify the intention and make sure they match the logic that follows. `block_name` isn't used in the code block below.